src/routes/teacher.route.js

```diff
@@ -4,10 +4,11 @@ import { authJwt } from '../services/auth.service.js';
 
 const router = express.Router();
 
-router.route('/api/teachers').get(authJwt, teacherCtrl.list);
+router.route('/teachers').get(authJwt, teacherCtrl.list);
 
 router
-    .route('/api/teachers/:teacherId')
+    .route('/teachers/:teacherId')
+    .get(teacherCtrl.readPublic)
```

Comment:
Can't have two GET handlers under the same route path.  Creating a route specifically for public would work, something like:

```javascript
router.route('/teachers/:teacherId/public').get(teacherCtrl.readPublic)
```